KinEcosystem/Core/Kin.swift

```diff
@@ -256,4 +273,67 @@ public class Kin {
     func closeMarketPlace() {
         mpPresentingController?.dismiss(animated: true, completion: nil)
     }
+    
+    fileprivate func setupBIProxies() {
+        EventsStore.shared.userProxy = UserProxy(balance: { [weak self] () -> (Double) in
+            guard let balance = self?.core?.blockchain.lastBalance else {
+                return 0
+            }
+            return NSDecimalNumber(decimal: balance.amount).doubleValue
+            }, digitalServiceID: { [weak self] () -> (String) in
+                guard let appId = self?.core?.network.client.authToken?.app_id else {
+                    if let startAppid = self?.core?.network.client.config.appId {
+                        return startAppid
+                    }
+                    return ""
+                }
+                return appId
+            }, digitalServiceUserID: { [weak self] () -> (String) in
+                guard let uid = self?.core?.network.client.authToken?.user_id else {
+                    if let startUid = self?.core?.network.client.config.userId {
+                        return startUid
+                    }
+                    return ""
+                }
+                return uid
+            }, earnCount: { () -> (Int) in
+                0
+        }, entryPointParam: { () -> (String) in
+            ""
+        }, spendCount: { () -> (Int) in
+            0
+        }, totalKinEarned: { () -> (Double) in
+            0
+        }, totalKinSpent: { () -> (Double) in
+            0
+        }, transactionCount: { () -> (Int) in
+            0
+        })
+        
+        EventsStore.shared.clientProxy = ClientProxy(carrier: { [weak self] () -> (String) in
+            return self?.bi.networkInfo.subscriberCellularProvider?.carrierName ?? ""
+            }, deviceID: { () -> (String) in
+                DeviceData.deviceId
+            }, deviceManufacturer: { () -> (String) in
+                "Apple"
+        }, deviceModel: { () -> (String) in
+            UIDevice.current.model
+        }, language: { () -> (String) in
+            Locale.autoupdatingCurrent.languageCode ?? ""
+        }, os: { () -> (String) in
+            UIDevice.current.systemVersion
+        })
+        
+        EventsStore.shared.commonProxy = CommonProxy(eventID: { () -> (String) in
+            UUID().uuidString
+        }, platform: { () -> (String) in
+            "iOS"
+        }, timestamp: { () -> (String) in
+            "\(Date().timeIntervalSince1970)"
+        }, userID: { [weak self] () -> (String) in
+            self?.core?.network.client.authToken?.ecosystem_user_id ?? ""
+            }, version: { () -> (String) in
+                "0.4.6"
```

Comment:
"0.4.6" should be taken from one place

Reply:
true - the sdk needs a normal versioning system - will open an issue for it.